integration_tests/run.sh

```
@@ -1,27 +1,157 @@
+    fi
+  done <<EOF
+$(list_protocols "$mode")
+EOF
+}
+
+build_parallel_groups() {
+  local width="$1"
+  local idx=0
+  local group=()
+
+  if [ "${#parallel_protocols[@]}" -eq 0 ]; then
```

Shellcheck (0.11.0)

[warning] parallel_protocols is referenced but not assigned.

(SC2154)